src/test/regress/expected/multi_read_from_secondaries.out

```diff
@@ -75,12 +75,95 @@ DEBUG:  Plan XXX query after replacing subqueries and CTEs: SELECT a FROM (SELEC
 ---------------------------------------------------------------------
 (0 rows)
 
+-- intermediate result pruning should still work
```

Comment:
Minor: can we insert rows before switching to secondary?

Reply:
Added some data and updated the test